### PR TITLE
fix(emotion-utils):  refactor position function

### PR DESCRIPTION
**Pull request by yoonminsang**
## Overview

- improve type safe
- change wrong variable name

## PR Checklist

- [x] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/react/emotion-utils/src/utils/position.ts [hunk 1]
```diff
@@ -71,14 +71,19 @@ export function position(position: CSSProperties['position'], coordinates?: Coor
...
-    : [positionOrTop, topOrCoordinates as CSSPixelValue, ...values];
+    ? isCSSPixelValue(topOrRightOrCoordinates)
+      ? [topOrRightOrCoordinates, ...values]
+      : [
+          topOrRightOrCoordinates.top,
+          topOrRightOrCoordinates.right,
+          topOrRightOrCoordinates.bottom,
+          topOrRightOrCoordinates.left,
+        ]
+    : [positionOrTop, topOrRightOrCoordinates as CSSPixelValue, ...values];
```
Comment (HoseungJang) [inline packages/react/emotion-utils/src/utils/position.ts:86]: These changes look just renaming. I think it's far from refactoring.
How about making the code more intuitive? (e.g. using IIFE to group related codes, or adding some comments to note what this code means, etc.)

**Comment by HoseungJang** [comment]
I'll close this PR. Note my comment please. Thank you @yoonminsang !

**Comment by yoonminsang** [comment]
Thanks for comments  @HoseungJang 

- rename

I think renaming is refactoring. 
I am studying refactoring with refactoring2 book(Martin Fowler). The book say naming is important. I attach my github link.
[my github repo link](https://github.com/yoonminsang/refactoring-2/blob/main/yoonminsang-robin/3/README.md#%EA%B8%B0%EC%9D%B4%ED%95%9C-%EC%9D%B4%EB%A6%84)

First I read this code, `topOrCoordinates` means top or Coordinates. But it means top or right or Coordinates. So I think this naming is wrong. If `topOrRightOrCoordinates` is so long, I rewrite something.

- IIFE

I think `Nested Ternaries` is poor readability. But it taste difference. I attach links.
[nested-ternaries-are-great](https://medium.com/javascript-scene/nested-ternaries-are-great-361bddd0f340)
[nested-ternaries-are-bad](https://medium.com/@nimelrian/no-no-no-152f46558b48)

If you agree with nested ternaries are great, I will pull request this code.
```
  const [top, right, bottom, left] = (() => {
    if (!isPositionValue(positionOrTop)) {
      return [positionOrTop, topOrRightOrCoordinates as CSSPixelValue, ...values];
    }
    if (!isCSSPixelValue(topOrRightOrCoordinates)) {
      return [
        topOrRightOrCoordinates.top,
        topOrRightOrCoordinates.right,
        topOrRightOrCoordinates.bottom,
        topOrRightOrCoordinates.left,
      ];
    }
    return [topOrRightOrCoordinates, ...values];
  })();
```

- Other commits
I write 3 commits. You comment only one commit. Other commits is ok?

If you want a new pull request, I will pull request. Thank you.

**Comment by HoseungJang** [comment]
> I think renaming is refactoring.
I am studying refactoring with refactoring2 book(Martin Fowler). The book say naming is important. I attach my github link.

Yes, you are absolutely right. Sometimes renaming would be main point to make a readable code.

But, in the case of this `position` function, I think just renaming `topOrCoordinates` to `topOrRightOrCoordinates` have no impact to make it more readable.

But your IIFE example is what I want. It would be more readable compared to complex ternary. Please apply it and reopen this PR. Thanks.

**Pull request closed without merging**